src/utils/setLocaleStorage.js

```diff
@@ -0,0 +1,7 @@
+export const setLocaleStorage = (key, data) => {
+  try {
+    const jsonData = JSON.stringify(data);
+
+    window.localStorage.setItem(key, jsonData);
+  } catch (error) {}
```

Comment:
It's generally a bad practice to have an empty catch block. Even if you think that an error here should not break the application, you should at least log it to the console. This will help you to debug issues and understand what went wrong.